### PR TITLE
Restructure server code under cloud.

### PR DESCRIPTION
**Pull request by shaper**
Move top level Dockerfile and cloudbuild.yaml under cloud directory
for better encapsulation.

Update cloud npm modules typescript and esm to match top level version.

Docker image and rollup are broken as part of this move. Fixing was
more complex than minor path edits, and this incremental step brings
top level cleanup benefits for the broader team/project.

README.md updated with a note at top of file to reflect this status
for the time being.

### CODE REVIEW

File: cloud/deployment/environment.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+export PROJECT_ID=arcs-project-223901
```
Comment (alxrsngrtn) [inline cloud/deployment/environment.sh]: Consider making these default values / allow developers to set their own env variables.

https://github.com/dylanaraps/pure-sh-bible#default-value

File: cloud/js-green-licenses.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    /* WTFPL */
+    "utf8-byte-length",
+    /* WTFPL */
+    "argsarray",
+    /* WTFPL */
+    "expand-template",
+    /* WTFPL */
+    "spark-md5",
+    /* Apache 2.0, but not discoverable */
+    "@tensorflow/tfjs-node",
```
Comment (alxrsngrtn) [inline cloud/js-green-licenses.json]: We should consider removing this for now? #3330

Comment (shaper) [reply cloud/js-green-licenses.json]: Done.

File: cloud/deployment/environment.sh (hunk 1 above)
Comment (shaper) [reply cloud/deployment/environment.sh]: Added TODO(cromwellian) @cromwellian

**Pull request merged**